src/core/core.scale.js

```diff
@@ -425,13 +438,7 @@ var Scale = Element.extend({
 	buildTicks: helpers.noop,
 	afterBuildTicks: function(ticks) {
 		var me = this;
-		// ticks is empty for old axis implementations here
-		if (helpers.isArray(ticks) && ticks.length) {
-			return helpers.callback(me.options.afterBuildTicks, [me, ticks]);
-		}
-		// Support old implementations (that modified `this.ticks` directly in buildTicks)
-		me.ticks = helpers.callback(me.options.afterBuildTicks, [me, me.ticks]) || me.ticks;
-		return ticks;
+		return helpers.callback(me.options.afterBuildTicks, [me, ticks]) || ticks;
```

Comment:
Should use `this` here, since its only used twice.

Reply:
done